src/pages/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,158 @@
+    const [showDialog, setShowDialog] = useState(false);
+    const [currentDialog, setCurrentDialog] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const [filters, setFilters] = useState({
+        searchQuery: "",
+        selectedGenre: "",
+        selectedSortControl: "release_date",
+    });
+
+    async function fetchMovies(filters: any, signal: AbortSignal) {
+        const response = await axios.get(
+            `http://localhost:4000/movies?limit=12&searchBy=title&search=${filters.searchQuery}&filter=${filters.selectedGenre}&sortBy=${filters.selectedSortControl}&sortOrder=asc`,
+            { signal }
+        );
+        return response.data.data;
+    }
+
+    useEffect(() => {
+        const controller = new AbortController();
+        const signal = controller.signal;
+
+        async function getMoviesList() {
+            setIsLoading(true);
+            try {
+                const movies = await fetchMovies(filters, signal);
+                setMovieList(movies);
+            }
+            catch (error) {
+                if (axios.isAxiosError(error)) {
+                    if (error.name === "Canceled error") {
+                        console.log("Previous request has been canceled: ", error.message);
+                    } else {
+                        console.log("Axios Error: ", error.message);
+                    }
+                } else {
+                    console.error("Error getting movie list: ", error);
+                }
+            } finally {
+                setIsLoading(false);
+            }
+        }
+
+        getMoviesList();
+
+        return () => {
+            controller.abort();
+        };
+    }, [filters]);
+
+    function updateFilters(key: "searchQuery" | "selectedGenre" | "selectedSortControl", value: string) {
+        setFilters((prevState) => ({
+            ...prevState,
+            [key] : value,
+        }));
+    }
+
+    function handleSearch(searchInput: string) {
+        updateFilters("searchQuery", searchInput);
+    }
+
+    function handleGenreSelect(genre: string) {
+        updateFilters("selectedGenre", genre === "All" ? "" : genre);
+    }
+
+    function handleSortControlChange(option: string) {
+        updateFilters("selectedSortControl", option);
+    }
+
+    function handleTileClick(movieDetails: MovieDetailsData) {
+        setSelectedMovie(movieDetails);
+    }
+
+    function handleToggleDialog() {
+        setShowDialog(!showDialog);
+    }
+
+    function handleShowDialog(dialogOption: string, movieDetails?: MovieDetailsData) {
+        setCurrentDialog(dialogOption);
+
+        if (movieDetails) {
+            setSelectedMovie(movieDetails);
+        }
+
+        handleToggleDialog();
+    }
+
+    function handleSubmit(MovieData: MovieDetailsData) {
+        console.log(MovieData);
+    }
+
+    function handleBackToSearch() {
+        setSelectedMovie(null);
+    }
+
```

Comment:
You have such many functions here. Try to think and somehow optimize it. Maybe take a look into useReducer hook and replace useState with this object.